pkg/db/factory.go

```diff
@@ -36,6 +41,28 @@ func WithMigrationTimeout(timeout time.Duration) FactoryOption {
 	}
 }
 
+func WithEncryptionConfiguration(ctx context.Context, configPath string) (FactoryOption, error) {
+	encryptionConf, err := encryptionconfig.LoadEncryptionConfig(ctx, configPath, false)
+	if err != nil {
+		return nil, err
+	}
+
+	// Although the config reading code expects GroupResources, we expect GroupKinds,
+	// so just convert, assuming that the Resource is actually a Kind, but lowercase.
+	transformers := make(map[schema.GroupKind]value.Transformer, len(encryptionConf.Transformers))
+	for gr, t := range encryptionConf.Transformers {
+		if len(gr.Resource) < 2 {
+			return nil, fmt.Errorf("invalid Resource: %s", gr.Resource)
+		}
+
+		transformers[schema.GroupKind{Group: gr.Group, Kind: strings.ToUpper(gr.Resource[:1]) + gr.Resource[1:]}] = t
```

Comment:
I think we should just take the resource string as is. We just document that resource in the config file should be populated with Kind.

Reply:
Unfortunately, the code that reads the config file returns an error if the resource name is capitalized. So it needs to remain lowercase, or I can copy and adapt that function.